### PR TITLE
Adds .vscode folder and upgrades docusaurus.

### PR DESCRIPTION
**Pull request by enigmalea**
- Initial commit on this PR upgrades docusaurus and caniuse-lite definitions
- second commit adds .vscode folder to all sharing of code snippets and make vscode extension suggestions.

### CODE REVIEW

File: .vscode/codesnippets.code-snippets [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		"prefix": "Code Block",
+		"body": [
+			"```bash showLineNumber",
+			"$1",
+			"```"
+		],
+		"description": "Create a shell code block with line numbers visible"
+	},
+
+	"Reading Time": {
```
Comment (essential-randomness) [inline .vscode/codesnippets.code-snippets:84]: I think reading time is a docusaurus plugin, and I'd discourage adding it manually

**Review by essential-randomness — APPROVED** [review]
Aside from the comment on reading time, the rest looks good to me

File: .vscode/codesnippets.code-snippets (hunk 1 above)
Comment (enigmalea) [reply .vscode/codesnippets.code-snippets:84]: It's only a docusaurus plug-in on their blog posts. This is just a markdown code snippet to add these things I used to denote how long external pages would take to read in the "How to Install WSL" page.

image

Now that that page is complete, I can remove it, if we don't want to do time estimates on other things. We can also remove time estimates on the WSL page, but I received feedback that the page was intimidating with a lot of external links, so I added the fact that they're only a few minutes each to try to combat that.

Comment (essential-randomness) [reply .vscode/codesnippets.code-snippets:84]: Ooooh makes sense. I think we can keep it then, but maybe just be explicit in the description like "add an inline reading time annotation for external links" (or something), so people know it's not the same functionality as the global plugin I was thinking about.

**Review by essential-randomness — APPROVED** [review]

File: .vscode/codesnippets.code-snippets (hunk 1 above)
Comment (enigmalea) [reply .vscode/codesnippets.code-snippets:84]: Pushing this update now!

**Review by essential-randomness — APPROVED** [review]

**Pull request merged**